admin/command_runner_test.go

```diff
@@ -66,21 +76,17 @@ func (suite *CommandRunnerSuite) SetupCommandRunner(opts ...CommandRunnerOption)
 	suite.cancel = cancel
 
 	signalerCtx, errChan := irrecoverable.WithSignaler(ctx)
+	go unittest.NoIrrecoverableError(ctx, suite.T(), errChan)
+
+	suite.grpcAddressSock = fmt.Sprintf("%s/%s-flow-node-admin.sock", os.TempDir(), unittest.GenerateRandomStringWithLen(16))
+	opts = append(opts, WithGRPCAddress(suite.grpcAddressSock))
 
 	logger := zerolog.New(zerolog.NewConsoleWriter())
 	suite.runner = suite.bootstrapper.Bootstrap(logger, suite.httpAddress, opts...)
 	suite.runner.Start(signalerCtx)
 	<-suite.runner.Ready()
-	go func() {
-		select {
-		case <-ctx.Done():
-			return
-		case err := <-errChan:
-			suite.Fail("encountered unexpected error", err)
-		}
-	}()
```

Comment:
I switched it to use `go unittest.NoIrrecoverableError()`

Reply:
I still don't know what you mean

Reply:
I actually refactored since that last comment. Turns out using a separate goroutine is unreliable for capturing the error. Instead, there's a new mocked `SignallerContext` implementation that doesn't use a separate goroutine and instead fails the test directly when `Throw` is called. See `irrecoverable.NewMockSignalerContext`